org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/MavenModelManager.java

```diff
@@ -146,6 +147,34 @@ public org.apache.maven.model.Model readMavenModel(File pomFile) throws CoreExce
     return null;
   }
 
+  public Optional<File> locatePom(File baseDir) {
```

Comment:
Please add @since and javadoc for this new API.

Reply:
I might move this to another place instead here ...